src/h.js

```diff
@@ -1,22 +1,29 @@
-export function h(tag, data) {
-  var node
-  var stack = []
-  var children = []
+function addChild(children, val) {
+  if (val == null || typeof val === 'boolean') return
 
-  for (var i = arguments.length; i-- > 2; ) {
-    stack[stack.length] = arguments[i]
+  if (Array.isArray(val)) {
+    for (var j = 0; j < val.length; j++) {
```

Comment:
Yes, that's an optimization for the _iterated_ function & that's why I intentionally left it out. If you replace the array-push with *any* other function, the cached loop is (still) always faster:

```js
  var i=0, len=data.length;
  for (; i < len; i++) {
    (6 + 1);
  }
```

```
cached loop---
elapsed:  440
op/s  2272727272.7272725
non-cached loop---
elapsed:  457
op/s  2188183807.439825
```

Again, these are still microbenchmarks, and mean nothing on their own. You should run the benchmark suite after applying the change & look for changes. This change has benefitted 99.999% of my real-world apps so I'm fairly confident in it. 😜